test/testcondition.cpp

```diff
@@ -4760,6 +4760,21 @@ class TestCondition : public TestFixture {
               "}");
         ASSERT_EQUALS("", errout_str());
 
+        check("struct S {\n"
+              "    const S* get2() const {\n"
+              "        if (mS)\n"
+              "            return mS;\n"
+              "        return this;\n"
+              "    }\n"
+              "    S* mS = nullptr;\n"
```

Comment:
Would this also work if `mS` was initialized in a constructor?

Reply:
It doesn't matter what `mS` was initialized to, `get2()` will always return a non-null pointer.